app/controllers/accounts_controller.rb

```diff
@@ -41,18 +37,20 @@ def destroy
     redirect_to root_path, notice: 'Account was successfully deleted.'
   end
 
-private
+  private
 
   def set_account
     @account = current_user.accounts.find(params[:id])
+  rescue ActiveRecord::RecordNotFound
+    account_not_found
+  end
```

Comment:
You are rescuing from this exception in application controller. No need to have it here as well.